File: TravelWise-backend/src/main/java/com/example/travelwise/entity/Category.java
package com.example.travelwise.entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private String icon;
    @ManyToOne
    @JoinColumn(name="department_id")
    private Department department;
}

File: TravelWise-frontend/src/app/models/category.ts
export class Category {
  id?: number;
  name: string;
  description: string;
  icon: string;
  departmentId: number;

  constructor(name: string, description: string, icon: string, departmentId: number) {
    this.name = name;
    this.description = description;
    this.icon = icon;
    this.departmentId = departmentId;
  }
}
